subgraph/src/subscriptions.ts

```diff
@@ -128,20 +124,18 @@ export function handleUnsubscribe(event: UnsubscribeEvent): void {
   entity.user = user.id;
   entity.save();
 
-  let sub = ActiveSubscription.load(event.params.user);
+  let sub = Subscription.load(event.params.user);
+  if (sub == null) return;
+
   // To handle an edge-case where the Subscribe/Unsubscribe events aren't received by the subgraph mapping in the same order they are emitted,
   // if a `UserSubscriptionCreatedEvent` exists in the same timestamp, don't create the `UserSubscriptionCanceledEvent` record
-  let createdEventId = buildUserSubscriptionEventId(
-    user.id,
-    USER_SUBSCRIPTION_EVENT_TYPE__CREATED,
-    event.block.timestamp
-  );
-  let createdEvent = UserSubscriptionCreatedEvent.load(createdEventId);
-  if (sub != null && createdEvent != null) {
-    buildAndSaveUserSubscriptionCanceledEvent(user, sub, event);
-  }
+  let subscribeEvent = Subscribe.load(event.transaction.hash);
+  if (subscribeEvent != null) return;
```

Comment:
It looks like the `id` of the `Subscribe` entity is the hash + log index:
```ts
 let entity = new Subscribe(
    event.transaction.hash.concatI32(event.logIndex.toI32())
  );
```
So I assume we need to add that here to find the `Subscribe` as well?

Reply:
Good catch. Thanks.

I'm wondering if we even need that edge-case handled, or should we just believe Leo.